dashboard/app/cache.go

```diff
@@ -133,3 +138,71 @@ func (c *CachedBugStats) Record(bug *Bug) {
 func cacheKey(ns string, accessLevel AccessLevel) string {
 	return fmt.Sprintf("%v-%v", ns, accessLevel)
 }
+
+func CachedBugGroups(c context.Context, ns string, accessLevel AccessLevel) ([]*uiBugGroup, error) {
+	item, err := memcache.Get(c, cachedBugGroupsKey(ns, accessLevel))
+	if err == memcache.ErrCacheMiss {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	jsonData, destructor := image.MustDecompress(item.Value)
+	defer destructor()
+
+	var ret []*uiBugGroup
+	err = json.Unmarshal(jsonData, &ret)
+	return ret, err
+}
+
+func cachedBugGroupsKey(ns string, accessLevel AccessLevel) string {
+	return fmt.Sprintf("%v-%v-bug-groups", ns, accessLevel)
+}
+
+// minuteCacheUpdate updates memcache every minute (called by cron.yaml).
+func minuteCacheUpdateHandler(w http.ResponseWriter, r *http.Request) {
+	c := appengine.NewContext(r)
+	for ns, nsConfig := range getConfig(c).Namespaces {
+		if !nsConfig.CacheMainPage {
+			continue
+		}
+		err := minuteCacheUpdate(c, ns)
+		if err != nil {
+			http.Error(w, fmt.Sprintf("bug groups cache update for %s failed: %v", ns, err),
+				http.StatusInternalServerError)
+			return
+		}
+	}
+}
+
+func minuteCacheUpdate(c context.Context, ns string) error {
+	bugs, _, err := loadAllBugs(c, nil)
```

Comment:
Let's "loadVisibleBugs" instead to follow the handleMain() logic.
I expected minuteCacheUpdate() to call handleMain() with minimal changes on the handleMain() side.
The cache read attempt was the only expected change in handleMain().

Reply:
I've changed it to `loadVisibleBugs` and adjusted the test to verify we only return bugs from the same namespace.